### PR TITLE
Binary Heap

### PR DESCRIPTION
**Pull request by shivasinghal2000**
Added Binary Heap

### CODE REVIEW

**Comment by shivasinghal2000** [comment]
labels were not added

File: Data Structures/ReadMe.md [hunk 1]
```diff
@@ -1 +1,36 @@
-Data Structures part of the repository
```
Comment (frextrite) [inline Data Structures/ReadMe.md]: @shivasinghal2000 Having this README *exactly the same* as the top-level README doesn't make much sense. Please revert this change or alternately you can write 1-2 sentences about data structures 😄

Comment (shivasinghal2000) [reply Data Structures/ReadMe.md]: @frextrite okay i will do all the required changes
